src/main/java/org/apache/tika/parser/external/WaitingExternalParser.java

```diff
@@ -15,48 +13,362 @@
+                    xhtml, metadata, tmp);
+        } finally {
+            tmp.dispose();
+        }
+    }
+
+    private void parse(
+            TikaInputStream stream, XHTMLContentHandler xhtml,
+            Metadata metadata, TemporaryResources tmp)
+            throws IOException, SAXException, TikaException {
+        boolean inputToStdIn = true;
+        boolean outputFromStdOut = true;
+        boolean hasPatterns = (metadataPatterns != null && !metadataPatterns.isEmpty());
+
+        File output = null;
+
+        // Build our command
+        String[] cmd = new String[command.length];
+        System.arraycopy(command, 0, cmd, 0, command.length);
+        for(int i=0; i<cmd.length; i++) {
+           if(cmd[i].indexOf(INPUT_FILE_TOKEN) != -1) {
+              cmd[i] = cmd[i].replace(INPUT_FILE_TOKEN, stream.getFile().getPath());
+              inputToStdIn = false;
+           }
+           if(cmd[i].indexOf(OUTPUT_FILE_TOKEN) != -1) {
+              output = tmp.createTemporaryFile();
+              outputFromStdOut = false;
+           }
+        }
+
+        // Execute
+        Process process;
+        if(cmd.length == 1) {
+           process = Runtime.getRuntime().exec( cmd[0] );
+        } else {
+           process = Runtime.getRuntime().exec( cmd );
+        }
+
+        Thread stdoutParser = null;
+        Thread stderrParser = null;
+        try {
+            if(inputToStdIn) {
+               sendInput(process, stream);
+            } else {
+               process.getOutputStream().close();
+            }
+
+            InputStream out = process.getInputStream();
+            InputStream err = process.getErrorStream();
+            
+            if(hasPatterns) {
+               stderrParser = extractMetadata(err, metadata);
+               
+               if(outputFromStdOut) {
+                  extractOutput(out, xhtml);
+               } else {
+                  stdoutParser = extractMetadata(out, metadata);
+               }
+            } else {
+               ignoreStream(err);
+               
+               if(outputFromStdOut) {
+                  extractOutput(out, xhtml);
+               } else {
+                  ignoreStream(out);
+               }
+            }
+        } finally {
+            try {
+                process.waitFor();
+            } catch (InterruptedException ignore) {
+            }
+            
+            waitForParserToFinish(stdoutParser);
+            waitForParserToFinish(stderrParser);
+        }
+
+        // Grab the output if we haven't already
+        if (!outputFromStdOut) {
+            extractOutput(new FileInputStream(output), xhtml);
+        }
+    }
+    
+    private void waitForParserToFinish(Thread parser)
+    {
+        if (parser != null)
+        {
+            try 
+            {
+                parser.join();
+            }
+            catch (InterruptedException e)
+            {
+                e.printStackTrace();
```

Comment:
This would be thrown when the process was asked to stop so we don't need to print stack trace.

We may want to call `parser.interrupt()` though.  I don't recall if the interrupt state propagates to spawned new threads like parser here.